### PR TITLE
fix: Refactored wrapper to load and wrap handler early

### PR DESCRIPTION
**Pull request by mrickard**
This PR refactors the Node wrapper file and updates tests. [A previous PR](https://github.com/newrelic/newrelic-lambda-layers/pull/127) introduced support for no-code-change instrumentation of ESM functions, but handled CJS functions asynchronously, which was not compatible with callback-returning Lambda functions. [A later PR](https://github.com/newrelic/newrelic-lambda-layers/pull/156) resolved that issue, but deferred wrapping the user's Lambda handler until invocation time. This PR wraps handlers on load rather than on invocation, and preserves a synchronous path for CJS functions along with an asynchronous path for ESM functions.

### CODE REVIEW

**Review by jsumners-nr — COMMENTED** [review]
Edit: it seems this is being handled in some other way. Disregard this request.

Can we update the following lines to include `npm install --production`?:

https://github.com/newrelic/newrelic-lambda-layers/blob/4c9d718c3203b3ecc883b7f5a8f888becdbd8e5d/nodejs/publish-layers.sh#L24

https://github.com/newrelic/newrelic-lambda-layers/blob/4c9d718c3203b3ecc883b7f5a8f888becdbd8e5d/nodejs/publish-layers.sh#L37

https://github.com/newrelic/newrelic-lambda-layers/blob/4c9d718c3203b3ecc883b7f5a8f888becdbd8e5d/nodejs/publish-layers.sh#L72

https://github.com/newrelic/newrelic-lambda-layers/blob/4c9d718c3203b3ecc883b7f5a8f888becdbd8e5d/nodejs/publish-layers.sh#L85

**Review by jsumners-nr — APPROVED** [review]
The only thing I have enough context to comment on is some effective globals in the tests, but I wouldn't block for that. Overall, it looks good to me.

File: nodejs/test/unit/cjsHandler.tap.js [hunk 1]
```diff
@@ -27,11 +29,8 @@ tap.test('Layer Handler - CJS Function', (t) => {
     process.env = { ...originalEnv }
     helper.unload()
   })
-  
-  t.test('should wrap handler in transaction', async(t) => {
-    process.env.NEW_RELIC_LAMBDA_HANDLER = 'test/unit/fixtures/cjs/handler.handler'
-    process.env.AWS_LAMBDA_FUNCTION_NAME = 'testFn'
 
+  t.test('should wrap handler in transaction', async(t) => {
```
Comment (bizob2828) [inline nodejs/test/unit/cjsHandler.tap.js:33]: do we need a unit test for lambda in cjs with callback or are we just relying on the integration tests for this?

**Review by bizob2828 — APPROVED** [review]
overall it looks good. One question but nice work refactoring this 👏🏻

**Pull request merged**